osmdroid-android/src/main/java/org/osmdroid/views/overlay/Overlay.java

```diff
@@ -300,26 +318,90 @@ protected synchronized static void drawAt(final Canvas canvas, final Drawable dr
         canvas.restore();
     }
 
-    /**
-     * Triggered on application lifecycle changes, assuming the mapview is triggered appropriately
-     * related issue https://github.com/osmdroid/osmdroid/issues/823
-     * https://github.com/osmdroid/osmdroid/issues/806
-     *
-     * @since 6.0.0
-     */
-    public void onPause() {
```

Comment:
removing or renaming a public api with hundreds of custom implementations is a horrendously bad idea

Reply:
Something have to be changed starting from somehow or we will continue to use a 10 years ago "mentality" where Devs have to call "onResume/onPause" methods, valiables have to be "nullified" just because 10+ years ago the Garbage Collector was not working well as today (nullify is not more required from years), @Nullable/@NotNull/@UiThread/@MainThread and other Annotations are never used to help Dev to understand things better (Android Studio notice things based on those useful Annotations) and so on.
We can't stay at code writte in 2000 year style....

I saw that many things I could changed and "modernized" but I choosed to left them as they were and changed just the minimum parts to achieve my purposes that I explained in the presentation post.
This library is good but from the today-POV it is very "obsolete" and written in a Java-style of 10/15+ years ago that are not more usable/valid for newer Java standards.